### PR TITLE
EVG-5371 create config from a command

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

File: trigger/project_triggers.go [hunk 1]
```diff
@@ -38,9 +38,19 @@ func TriggerDownstreamVersion(args ProcessorArgs) (*version.Version, error) {
...
 
 	// get the downstream config
-	config, err := makeDownstreamConfigFromFile(args.DownstreamProject, args.File)
-	if err != nil {
-		return nil, err
+	var config *model.Project
+	if args.File != "" {
+		config, err = makeDownstreamConfigFromFile(args.DownstreamProject, args.File)
+		if err != nil {
+			return nil, err
```
Comment (bsamek) [inline trigger/project_triggers.go]: wrap?

File: trigger/project_triggers.go [hunk 2]
```diff
@@ -38,9 +38,19 @@ func TriggerDownstreamVersion(args ProcessorArgs) (*version.Version, error) {
...
+		}
+	} else if args.Command != "" {
+		config, err = makeDownstreamConfigFromCommand(args.DownstreamProject, args.Command)
+		if err != nil {
+			return nil, err
```
Comment (bsamek) [inline trigger/project_triggers.go]: wrap?

File: trigger/project_triggers.go [hunk 3]
```diff
@@ -38,9 +38,19 @@ func TriggerDownstreamVersion(args ProcessorArgs) (*version.Version, error) {
...
+		}
+	} else {
+		return nil, errors.New("must specify a file xor command to define downstream project config")
```
Comment (tychoish) [inline trigger/project_triggers.go]: nit: s/xor/or/

File: trigger/project_triggers.go [hunk 4]
```diff
@@ -14,12 +15,11 @@ import (
 	"github.com/pkg/errors"
 )
 
+const TriggerCommandFileName = "evg-triggered-tasks.json"
```
Comment (tychoish) [inline trigger/project_triggers.go]: why is the name of the file/path to the file not also configurable?

File: trigger/project_triggers.go (hunk 1 above)
Comment (tychoish) [reply trigger/project_triggers.go]: then just do errors.WithStack()

File: trigger/project_triggers.go (hunk 2 above)
Comment (tychoish) [reply trigger/project_triggers.go]: errors.WithStack()

**Pull request merged**